### DISCUSSION TITLE
Replace camelcase fields with snakecase

### DISCUSSION THREAD

**Discussion by juancarlosjr97** [directus/directus#4683] [Feature Requests]
Adding camelcase, this is something personal because I fight with eslint but I believe it would be something nice instead of snake case fields. I understand as we are coming from PHP the snake case has to continue, but what about creating a parallel or a way to introduce camelcase to the system or migration that can be implemented if camelcase would be the path to continue.

**Comment by rijkvanzanten** [comment]
Snake-casing where specifically? If you're referring to the database; Directus just uses whatever's in the database. If you're talking about the system stuff specifically, it seems like generally speaking, snake-case is still often used for database modeling schemas and  JSON output for REST api's, so it's mostly a 🤷🏻 whatever's most common kinda approach

**Reply by juancarlosjr97** [reply]
Well, I am mainly thinking about JavaScript standards and GraphQL. I know it is more common to have GraphQL with camelcase and some of the most popular JS linter bring the camelcase standard, but I know it is not database standard, that is why I was thinking about a way to enable from camelcase to snakecase, without changing the current structure of the database as snake case.

What do you think?

**Reply by rijkvanzanten** [reply]
I think it comes down to personal preference 🤔 You're right about the JS standards (although there's no "rule" saying snake_case is forbidden of course), but we can't forget that Directus is used in all sorts of contexts (not just JS)

**Reply by juancarlosjr97** [reply]
Absolutely agree with you @rijkvanzanten 

Perhaps something that we can implement to the SDK directly, turning snake_case into camelcase, I don't think it would be that difficult as we only need to loop the response keys and change them, allowing to have camelcase to JS and we avoid changing the backend. I am actually doing that as my personal and work standard we cannot use camelcase, so I added a camelcase and snakecase converter when sending a request or receiving a request? Of course, it does not work with GraphQL but potentially could be the same implementation.

What do you think about that :)?

**Reply by rijkvanzanten** [reply]
I think it's a good idea for an npm library that you can run on any provided object, but I'm not too sure about having it included as part of the SDK 🤔

**Reply by rijkvanzanten** [reply]
Ha actually, it already exists! https://www.npmjs.com/package/camelcasify

I'd recommend using something like that then 👍🏻 

```js
const response = await directus.items("articles").read();

const result = camelcasify(response);
```

**Reply by juancarlosjr97** [reply]
Yes, it is good @rijkvanzanten :) 

Probably a good idea to build an Axios plugin to do this change automatically, an Axios interceptor.

**Comment by jaads** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
